pkg/catalogsourceconfig/update.go

```diff
@@ -62,10 +61,17 @@ func (r *updateReconciler) Reconcile(ctx context.Context, in *v1alpha1.CatalogSo
 // deleteObjects deletes the CatalogSource in the old TargetNamespace.
 func (r *updateReconciler) deleteObjects(in *v1alpha1.CatalogSourceConfig) error {
 	cachedCSCSpec, found := r.cache.Get(in)
-	// This should never happen as it is because the cached Spec has changed
+	// This generally won't happen as it is because the cached Spec has changed
 	// that we are in the update reconciler.
+	// However, in the case that the cache is invalidated because of a restart,
+	// we could simply be unsure if the targetNamespace changed. Ideally,  we
+	// we should be using something like a direct reference to the old object
+	// instead of an in memory cache, since when the service restarts we will be
+	// unable to find the old object if the targetNamespace changed while we were
+	// down. For now, we will just ignore it.
 	if !found {
```

Comment:
Question: I am trying to understand what real world scenario is causing this if block to be executed. We detected that the target namespace has changed ( `r.targetChanged` == true ) but the spec is not in cache.

Reply:
@tkashem because the cache invalidator logic changed. Previously we were cheating by saying that if we didn't find the `csc` in the cache that we should assume that everything is fine and ignore the cache. To be able to force a resync I updated that logic to assume that if the `csc` wasn't in the cache that the cache is not valid (which is an accurate assumption).

The problem is, we genuinely don't know if the targetNamespace was changed if we lose track of the cache. So we should say it's invalid and attempt to cleanup, but we don't have a way of finding the old one.

Reply:
Yeah, if the cache is missing then we can't answer the question whether the target namespace has changed. We will need to persist the namespaced name of `catsrc` in the future in `annotations` or something.